pipelines/shared/nextflow/color_workflow.nf

```diff
@@ -0,0 +1,202 @@
+
+process get_id_list {
+    publishDir params.final_output_dir, mode: 'copy'
+    input:
+        path cluster_id_map
+        path singletons
+        path seqid_source_map
+    output:
+        path 'id_lists/', emit: 'id_lists'
+        path 'cluster_sizes.txt', emit: 'cluster_sizes'
+    """
+    mkdir id_lists
+    mkdir id_lists/uniprot_ids id_lists/uniref90_ids id_lists/uniref50_ids
+    perl $projectDir/../shared/perl/get_id_lists.pl --cluster-map $cluster_id_map --singletons $singletons \
+        --uniprot id_lists/uniprot_ids --uniref90 id_lists/uniref90_ids --uniref50 id_lists/uniref50_ids \
+        --seqid-source-map $seqid_source_map --cluster-sizes cluster_sizes.txt \
+        --config ${params.efi_config} --db-name ${params.efi_db}
```

Comment:
Scripts should still be runnable without Nextflow, could you make Perl create these directories instead of doing it in the process? You could also output tuples instead of directory names so that each file is tagged with its uniref version.